File: GitNexus/gitnexus/test/fixtures/lang-resolution/csharp-overload-param-types/Models/UserService.cs
namespace Models;

public class User
{
    public string GetName() => "user";
}

public class UserService
{
    public User Lookup(int id)
    {
        return new User();
    }

    public User Lookup(string name)
    {
        return new User();
    }

    public void CallById()
    {
        Lookup(42);        // literal int → must resolve to Lookup(int) only
    }

    public void CallByName()
    {
        Lookup("alice");   // literal string → must resolve to Lookup(string) only
    }
}

File: GitNexus/gitnexus/test/fixtures/lang-resolution/java-overload-param-types/models/UserService.java
package models;

public class UserService {
    public User lookup(int id) {
        return new User();
    }

    public User lookup(String name) {
        return new User();
    }

    public void callById() {
        lookup(42);        // literal int → must resolve to lookup(int) only
    }

    public void callByName() {
        lookup("alice");   // literal String → must resolve to lookup(String) only
    }
}
